File: WorkoutApp-GraphQL/src/entities/Exercise.ts
export interface Exercise {
    id: string;
    name: string;
    primaryMuscle: Muscle;
    secondaryMuscle: Muscle;
    exerciseSteps: ExerciseStep[];
    equipment: Equipment[];
    exerciseCategorys: ExerciseCategory[];
}

export interface Muscle {
    id: string;
    name: string;
}

export interface ExerciseStep {
    exerciseStepNumber: number;
    description: string;
}

export interface Equipment {
    id: string;
    name: string;
}

export interface ExerciseCategory {
    id: string;
    name: string;
}

File: WorkoutApp.API/Models/Domain/Exercise.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace WorkoutApp.API.Models.Domain
{
    public class Exercise : IIdentifiable
    {
        public int Id { get; set; }
        [Required]
        [MaxLength(255)]
        public string Name { get; set; }
        public string Description { get; set; }
        public int? PrimaryMuscleId { get; set; }
        public int? SecondaryMuscleId { get; set; }
        public Muscle PrimaryMuscle { get; set; }
        public Muscle SecondaryMuscle { get; set; }
        public List<UserFavoriteExercise> FavoritedBy { get; set; }
        public List<ExerciseStep> ExerciseSteps { get; set; }
        public List<EquipmentExercise> Equipment { get; set; }
        public List<ExerciseCategoryExercise> ExerciseCategorys { get; set; }
    }
}